src/datastore/RestaurantDataStore.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * CS1D - Fast Food Associates
+ *
+ * Implements a data storage mechanism for Restaurant objects.
+ * These objects are contained in a doublly linked list that
+ * supports iterators and a comperable interfacen
+ *
+ * @author   edt
+ */
+
+#ifndef __RESTURANTDATASTORE_H_INCL__
+#define __RESTURANTDATASTORE_H_INCL__
+
+#include "MyDblLinkList.h"
+#include "Restaurant.h"
+#include <list>
+
+using namespace std;
+using namespace nsMyDblLinkList;
+
+class RestaurantDataStore
+{
+public:
+    // Constructor
+    RestaurantDataStore();
+
+    // Destructor
+    virtual ~RestaurantDataStore();
+
+    void printAsDebug(bool printeol, bool printcontent) const;
+
+    //std::list<Restaurant> list;
+    MyDblLinkList<Restaurant> list;
+
+private:
+
+    // Copy constructor
+    RestaurantDataStore(const RestaurantDataStore& src);

```

Comment:
If going for the singleton approach, better to just `= delete` this and `operator=`
```suggestion
    RestaurantDataStore(const RestaurantDataStore& src) = delete;
```

Reply:
Will "delete" erase all data that is in the database along with data that has been added by the admin (via Copy Constructor)?

Reply:
I'm pretty sure he means this line @ChrisRzech 

```suggestion
    RestaurantDataStore(const RestaurantDataStore& src)  = delete;;
```